### PR TITLE
Implement file downloading

### PR DESCRIPTION
**Pull request by OshanIvantha**
This will only start working after the PR #66 of js-owncloud-client is merged

### CODE REVIEW

**Comment by felixheidecke** [comment]
@OshanIvantha ping me if you need help resoling conflicts here.

**Comment by felixheidecke** [comment]
Well … downloading works, but not as expected. Can the download stream already start instead of loading the blob into the browser and then trigger the download?

Currently, I have to wait for the browser to load the file.

_I hope my words didn't confuse you_

File: apps/files/src/components/File-Details-Buttonrow.vue [hunk 1]
```diff
@@ -40,10 +40,31 @@
...
             return {
                 fileAction: null
             }
+        },
+        methods: {
+            download() {
+                console.log("cat")
+                console.log(this.files)
+                for(let i = 0; i < this.files.length; i++){
+                    OC.$client.files.getFileContents(this.files[i].path).then(res => {
```
Comment (DeepDiver1975) [inline apps/files/src/components/File-Details-Buttonrow.vue]: This is Just wrong. Let the browser handle the download.

File: apps/files/src/components/File-Details.vue [hunk 2]
```diff
@@ -1,131 +1,130 @@
 <template lang="pug">
-	// material icon mixin
```
Comment (DeepDiver1975) [inline apps/files/src/components/File-Details.vue]: Please fix the whitespace Messer. Thx

File: apps/files/src/components/File-Details-Buttonrow.vue (hunk 1 above)
Comment (OshanIvantha) [reply apps/files/src/components/File-Details-Buttonrow.vue]: Oh sorry I've missed some debug code.. 
Can you clarify on "Let the browser handle the download" please? I thought this is how we allow the download to happen through browser.

Comment (DeepDiver1975) [reply apps/files/src/components/File-Details-Buttonrow.vue]: Basically an anchor tag with the url to the File should be enough.

**Comment by OshanIvantha** [comment]
@felixheidecke I don't think the js-owncloud-client API supports download streaming. 
https://github.com/owncloud/js-owncloud-client

File: apps/files/src/components/File-Details-Buttonrow.vue (hunk 1 above)
Comment (OshanIvantha) [reply apps/files/src/components/File-Details-Buttonrow.vue]: That is what I have done. I have to call the *getFileContents* multiple times when the user has selected multiple files to be downloaded. As its current state, js-owncloud-client API doesn't support downloading multiple files with a single call. And then I let the browser handle each response using and anchor tag. 

If I am still not approaching that correctly, please advise.

File: apps/files/src/components/File-Details.vue (hunk 2 above)
Comment (OshanIvantha) [reply apps/files/src/components/File-Details.vue]: Fixed. File-Details.vue represents the same indentation in the master branch except for the lines I changed.

**Comment by felixheidecke** [comment]
> I don't think the js-owncloud-client API supports download streaming.

Hmm … we need a solution for this. @DeepDiver1975 is there any other way to request a file via direct link + token?

**Comment by felixheidecke** [comment]
Let's merge this for now and open a new PR concerning single and multi download.

**Comment by DeepDiver1975** [comment]
> Let's merge this for now and open a new PR concerning single and multi download.

the url to be used for downloading is created here https://github.com/owncloud/js-owncloud-client/blob/a9e48343b140dd1f6fb0678be44f0f590392ccde/owncloud/fileManagement.js#L78

and yes - looks like there is a method missing to expose this link ... lets add this ...
open question: how do we pass the credentials/token?

**Comment by DeepDiver1975** [comment]
oh ... seems like the solution is similar to what we already have implemented ... http://blog.bguiz.com/2014/07/03/file-download-with-http-request-header/

**Pull request merged**